src/components/eventsPageComponents/EventCard.jsx

```diff
@@ -91,6 +112,9 @@ function EventCard({ TheEvent }) {
                 <div className='flex justify-center text-[16px] shadow-md items-center bg-emerald-100 md:w-[30%] w-[50%] rounded-md lg:text-lg font-semibold'>
                     {formattedDate}
                 </div>
+                <div className='text-xl'>
```

Comment:
you didn't have to fetch the user document, you could"ve just used the context
userDoc={user.uid), this way you'll be making full use of the user context, actually you won't have to pass it even as an argument, you'll just call the useUser in the save event componenet, userContext was created for this reason

that way, code will be shorteer, and remember the first rule about useEffect," if you can avoid using use effect don't use it" - creater of react